test/edu/umass/cs/gnsclient/client/singletests/SelectGeoTest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ *
+ *  Copyright (c) 2015 University of Massachusetts
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you
+ *  may not use this file except in compliance with the License. You
+ *  may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ *  implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ *
+ *  Initial developer(s): Westy
+ *
+ */
+package edu.umass.cs.gnsclient.client.singletests;
+
+import edu.umass.cs.gnsclient.client.GNSClientCommands;
+import edu.umass.cs.gnsclient.client.util.GuidEntry;
+import edu.umass.cs.gnsclient.client.util.GuidUtils;
+import edu.umass.cs.gnsclient.client.util.JSONUtils;
+import edu.umass.cs.gnscommon.GNSProtocol;
+import edu.umass.cs.gnscommon.utils.RandomString;
+
+import java.awt.geom.Point2D;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+import static org.hamcrest.Matchers.*;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import static org.junit.Assert.*;
+
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+/**
+ * Functionality test for core elements in the client using the UniversalGnsClientFull.
+ *
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class SelectGeoTest {
```

Comment:
Yes, but I'm doing all those changes together as part of MOB-1084. This update is in service of MOB-1079 which is trying to find a bug. The addition of this file is to make a simple reproducible test case that  isolates the cause as much as possible. Although it probably doesn't make a difference in this case my general philosophy is to not introduce another variable (changing the existing code) until I've figure out what the problem is unless I think it's directly related to fixing the problem.